core/src/test/java/kafka/test/junit/ZkClusterInvocationContext.java

```diff
@@ -109,7 +109,9 @@ public void modifyConfigs(Seq<Properties> props) {
                     @Override
                     public Properties serverConfig() {
                         Properties props = clusterConfig.serverProperties();
-                        props.put(KafkaConfig.InterBrokerProtocolVersionProp(), metadataVersion().version());
+                        if (!props.containsKey(KafkaConfig.InterBrokerProtocolVersionProp())) {
+                            props.put(KafkaConfig.InterBrokerProtocolVersionProp(), metadataVersion().version());
```

Comment:
We used to override IBP to the latest version, which stops us from setting custom IBP using annotation.